### PR TITLE
Upgrade the csv-export JS package to the latest version

### LINKED ISSUE #7307: Transactions CSV file inserts tab character before any negative value
### Describe the bug
When downloading transactions to a CSV file, any fields with negative values (for example, the entire `Fees` column) have a _tab_ character added - this makes them hard to work with in programs such as Excel and Google sheets (interestingly, Apple's Numbers seems to ignore the tab character)

### To Reproduce
1. Go to `Payments > Transactions` on your site's wp-admin sidebar
3. Click on `Download`
4. Open the CSV file in Excel or Google sheets
5. try to do a `SUM` (or any other math function) with the `Fees` column, and it won't work as expected because of the tabs

### Screenshots
Screenshot 2023-09-27 at 10 11 07 AM
_showing the tab characters in the raw file by running a regex search in Visual Studio Code_

### Expected behavior
We should be able to do quick calculations within a spreadsheet program for all numerical values 

### Additional context
brought to my attention in 7086428-zen

### PR DESCRIPTION
**Pull request by anu-rock**
Fixes #7307

#### Changes proposed in this Pull Request

* Update the dev dependency version for the `@woocommerce/csv-export` package.
* Add the package to the dependencies document.

Note that the actual issue was fixed upstream at https://github.com/woocommerce/woocommerce/pull/41163.

#### Testing instructions

* No failing tests.
* Updated docs look good.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (not required)
- [ ] Tested on mobile (does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,7 +99,7 @@
     "@typescript-eslint/parser": "4.15.2",
     "@woocommerce/api": "0.2.0",
     "@woocommerce/components": "12.0.0",
-    "@woocommerce/csv-export": "1.7.0",
+    "@woocommerce/csv-export": "1.8.0",
```
Comment (anu-rock) [inline package.json:102]: May change once the package is [published](https://github.com/woocommerce/woocommerce/blob/trunk/tools/package-release/README.md).

**Comment by anu-rock** [comment]
Some tests are failing because a new version of upstream package has not yet been released. I'll come back for reviews once that happens.

File: package.json (hunk 1 above)
Comment (anu-rock) [reply package.json:102]: Confirming that 1.8.0 it is:  
https://www.npmjs.com/package/@woocommerce/csv-export/v/1.8.0

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7729` or branch name `update/7307-csv-export-package-version` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/7307-csv-export-package-version) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 63076604a6dbe1b04e71f61b2a88557bf5014f79
- Build time: 2023-11-28 08:19:07 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by deepakpathania — APPROVED** [review]
Changes LGTM, pre-approving before the tests pass.

**Pull request merged**